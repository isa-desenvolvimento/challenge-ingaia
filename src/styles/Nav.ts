import styled from 'styled-components'

export const Container = styled.div`
  width: 100%;
  margin: 0 auto;
  display: grid;
  justify-content: center;
  align-items: center;
  text-align: center;
`
export const InputContainer = styled.div`
  margin: 2rem 0;
  height: 3rem;
`
export const Input = styled.input`
  background-color: transparent;
  border-color: rgba(255, 255, 255, 0.664);
  color: white;
  padding-inline-start: 10px;
  border-radius: 5px;
  height: 100%;
  padding: 1.8rem;
`

// export const Button = styled.button`
//   margin-left: 15px;
//   width: 5em;
//   height: 100%;
//   color: white;
//   border-color: rgba(255, 255, 255, 0.664);
//   background-color: transparent;
//   border-radius: 5px;
//   transition-duration: 0.4s;
// `
