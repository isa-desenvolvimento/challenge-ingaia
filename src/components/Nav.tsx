import * as S from 'styles/Nav'
import { useRouter } from 'next/router'
import Image from 'next/image'
import { useState } from 'react'
import { useTranslationContext } from 'locales/translationContext'
import Button from 'components/Button'

const Nav = () => {
  const [search, setSearch] = useState('')
  const router = useRouter()
  const translation = useTranslationContext()

  const searchEvt = (e) => {
    if (e.target.value == '') {
      setSearch('')
    } else {
      setSearch(e.target.value)
    }
  }

  const handleSubmit = (e) => {
    e.preventDefault()

    if (e.target[0].form[0].value)
      router.push(`/characters/${e.target[0].form[0].value}`)
  }

  return (
    <S.Container>
      <Image src="/assets/image-logo.png" width="502" height="214" />

      <form onSubmit={handleSubmit}>
        <S.InputContainer>
          <S.Input
            placeholder="Search characters"
            value={search}
            onChange={(e) => searchEvt(e)}
          />
          <Button
            disabled={search === ''}
            type="submit"
            text={translation.SEARCH}
          />
        </S.InputContainer>
      </form>
    </S.Container>
  )
}

export default Nav
