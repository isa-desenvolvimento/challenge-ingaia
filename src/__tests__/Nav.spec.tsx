/* eslint-disable global-require */
import {
  render, fireEvent, act,
} from '@testing-library/react';
import router from 'next/router';
import Nav from '../components/Nav';

jest.mock('next/router', () => require('next-router-mock'));

describe('<Nav />', () => {
  it('should render component input', () => {
    const { getByPlaceholderText } = render(<Nav />);

    const inputNode = getByPlaceholderText(/Nunca dejes de buscar/i);
    expect(inputNode).toBeTruthy();
  });

  it('should render component Nav children', async () => {
    const value = 'apple';
    const { getByPlaceholderText, getByTestId } = render(<Nav />);

    act(() => {
      fireEvent.change(getByPlaceholderText(/Nunca dejes de buscar/i), { target: { value } });
      fireEvent.click(getByTestId('btn_nav'));

      router.push({
        pathname: `/items/${value}`,
      });

      expect(router).toMatchObject({
        pathname: '/items/apple',
      });
    });
  });

  it('should render snapshot', () => {
    const { container } = render(
      <Nav />,
    );
    expect(container).toMatchSnapshot();
  });
});
