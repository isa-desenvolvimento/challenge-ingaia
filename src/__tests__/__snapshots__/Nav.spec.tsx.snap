// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should render snapshot 1`] = `
<div>
  <nav
    class="container"
  >
    <div
      class="logo"
      data-testid="img_nav"
    />
    <form
      class="inputContainer"
      data-testid="frm_nav"
    >
      <input
        data-testid="inp_nav"
        placeholder="Nunca dejes de buscar"
        type="text"
      />
      <button
        class="search"
        data-testid="btn_nav"
        type="submit"
      >
         
      </button>
    </form>
  </nav>
</div>
`;
